src/software/ai/hl/stp/play/enemy_free_kick/enemy_free_kick_play_fsm.cpp

```diff
@@ -22,8 +22,16 @@ void EnemyFreeKickPlayFSM::blockEnemyKicker(const Update& event)
 
 void EnemyFreeKickPlayFSM::setTactics(const Update& event, unsigned int num_tactics)
 {
+    if (num_tactics <= 0)
+    {
+        return;
+    }
+
+    const double HALF_FIELD_LENGTH = event.common.world_ptr->field().totalYLength() / 2;
+    const Point FRIENDLY_GOAL_CENTER =
+        event.common.world_ptr->field().friendlyGoalCenter();
```

Comment:
you can move this into the struct under `private` and tag it as a `constexpr`.

Why do this?
Because `constexpr` get resolved during compilation and the compiler will directly replace uses of these variables directly with the number. This saves some performance costs.

Reply:
maybe not `FRIENDLY_GOAL_CENTER` but at least `HALF_FIELD_LENGTH`

also by out conventions, since this constant is local to this function `FRIENDLY_GOAL_CENTER`  doesn't have to be capitalized

Reply:
But both rely on the field length which is part of the world proto being sent in. Doesn't that mean the value cannot be determined at compile time?